internal/config/config_test.go

```diff
@@ -87,31 +102,14 @@ func TestConfig_init_ok(t *testing.T) {
 		description, inputs, rc, want := c.description, c.inputs, c.rc, c.want
 		t.Run(description, func(t *testing.T) {
 			tool := crt.Tool{
-				Name:     "thisaction",
-				Version:  "0.0.0",
-				Revision: "cabba9e",
+				Name:     "<tool-name>",
+				Version:  "<tool-version>",
+				Revision: "<tool-revision>",
 			}
 			got, err := inputs.init(rc, tool)
 			if err != nil {
 				t.Fatal(err)
 			}
-
-			// Test the verification build root separately because it's
-			// a temp directory with an unpredictable name.
-			if got.Verification.BuildRoot == "" {
-				t.Errorf("got empty VerificationBuildRoot")
-			}
-			// Force got and want build roots and build instructions to be empty so we
-			// can assert equality on everything else.
-			got.Verification = Paths{}
-			want.Verification = Paths{}
-			got.Primary = Paths{}
-			want.Primary = Paths{}
-			got.Parameters.Instructions = ""
-			want.Parameters.Instructions = ""
-			got.VerificationResult = ""
-			want.VerificationResult = ""
-
```

Comment:
This entire section used to exist because it was too hard to test some of the fields, because their values were dependent on the state of the source code and the current OS, so initially I opted not to test those fields. However, this PR adds the needed machinery to make these stable, by allowing overriding some functions that generate paths and IDs for testing purposes. Now we assert over the entire struct, so all the tests are stronger than they were.